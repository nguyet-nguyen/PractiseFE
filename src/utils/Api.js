import axios from 'axios';
console.log(process.env.REACT_APP_DOMAIN);
// config hosting-api here
export const baseURL = 'http://127.0.0.1/api';
export const api = axios.create({
    baseURL,
    headers: {
        'Content-Type': 'application/json',
    },
});

export const apiadmin = axios.create({
    baseURL,
    headers: {
        'Authorization': `Bearer ${window.localStorage.getItem('token')}`,
        "Content-Type": "multipart/form-data"
    },
});

export const apiadminnoFormdata = axios.create({
    baseURL,
    headers: {
        'Authorization': `Bearer ${window.localStorage.getItem('token')}`,
        'Content-Type': 'application/json',
    },
});
