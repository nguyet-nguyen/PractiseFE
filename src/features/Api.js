import {api, apiadmin, apiadminnoFormdata} from "../utils/Api";

// ---------------users------------------------?
export const getAllProducts = () => {
  try {
    return api.get("/products");
  } catch (error) {
    console.log(error);
  }
};

export const getAllCategory = () => {
  try {
    return api.get("/categories");
  } catch (error) {
    console.log(error);
  }
};
// get hot deal
export const getProductHotDeal = () => {
  try {
    return api.get("/bestSelling");
  } catch (error) {
    console.log(error);
  }
};

export const requestFilterCategory = (data) => {
  try {
    return api.post("/products/filter", data);
  } catch (error) {
    console.log(error);
  }
};

export const SignUpApi = (data) => {
  try {
    return api.post(`/register`, data);
  } catch (error) {
    console.log(error);
  }
};

export const SignInApiToken = (data) => {
  try {
    return api.post(`/login`, data);
  } catch (error) {
    console.log(error);
  }
};

export const SignInApiRole = (data) => {
  try {
    return api.post(`/users/email`, data);
  } catch (error) {
    console.log(error);
  }
};
// product detail
export const getProductDetail = (id) => {
  try {
    return api.get(`/products/${id}`);
  } catch (error) {
    console.log(error);
  }
};


// Add To Cart
export const addToCart = (data) => {
  try {
    return apiadmin.post(`/users/carts`, data);
  } catch (error) {
    console.log(error);
  }
};

// Get all items in cart by user
export const getAllItemsInCart = () => {
  try {
    return apiadmin.get(`/users/carts`);
  } catch (error) {
    console.log(error);
  }
};

// Remove item from shopping cart
export const removeItemFromCart = (id) => {
  try {
    return apiadmin.delete(`/users/carts/${id}`);
  } catch (error) {
    console.log(error);
  }
};

// Update quantity in shopping cart by cart id
export const updateCart = (id, data) => {
  try {
    return apiadmin.put(`/users/carts/${id}`, data);
  } catch (error) {
    console.log(error);
  }
};

// Checkout
export const addOrder = (data) => {
  try {
    return apiadminnoFormdata.post(`/users/orders`, data);
  } catch (error) {
    console.log(error);
  }
};

// Users order history
export const getUsersOrdHistory = () => {
  try {
    return apiadmin.get(`/users/orders`);
  } catch (error) {
    console.log(error);
  }
};

// ----------------------admin----------------------------------
export const getAllUsers = (page,limit) => {
  try {
    return apiadmin.get(`/admin/users?limit=${limit}&page=${page}`);
  } catch (error) {
    console.log(error);
  }
};

// create user
export const CreateUsers = (data) => {
  try {
    return apiadmin.post("/admin/users", data);
  } catch (error) {
    console.log(error);
  }
};

// add product
export const CreateProducts = (data) => {
  try {
    return apiadmin.post('/admin/products', data);
  } catch (error) {
    console.log(error);
  }
}

export const getAllProductsAdmin = (page,limit) => {
  try {
    return apiadmin.get(`/admin/products?limit=${limit}&page=${page}`);
  } catch (error) {
    console.log(error);
  }
}

export const DeleteProduct = (id) => {
  try {
    return apiadmin.delete(`/admin/products/${id}`);
  } catch (error) {
    console.log(error);
  }
}

export const UpdateProduct = (data,id) => {
  try {
    return apiadmin.post(`/admin/products/${id}`, data);
  } catch (error) {
    console.log(error);
  }
}
export const UpdateProductItems = (data,id) => {
  try {
    return apiadminnoFormdata.put(`/admin/products/${id}/productItem`, data);
  } catch (error) {
    console.log(error);
  }
}

// get all order list
export const getAllOrderListAdmin = (page,limit) => {
  try {
    return apiadmin.get(`/admin/orders?limit=${limit}&page=${page}`);
  } catch (error) {
    console.log(error);
  }
}
// update status orders list
export const UpdateStatusOrderList = (data,id) => {
  try {
    return apiadminnoFormdata.put(`/admin/orders/${id}`, data);
  } catch (error) {
    console.log(error);
  }
}
// report
export const getReports = () => {
  try {
    return apiadmin.get("/admin/reports");
  } catch (error) {
    console.log(error);
  }
};
// report chart
export const getReportsChart = () => {
  try {
    return apiadmin.get("/admin/reports/chart");
  } catch (error) {
    console.log(error);
  }
};
// ---------------------- for user and admin ----------------------------------

// Validate Email (check if email exists)
export const checkIfEmailExists = (data) => {
  try {
    return api.post("/users/check_email", data);
  } catch (error) {
    console.log(error);
  }
};

// Get User Info By Id 
export const getUserInfo = (id) => {
  try {
    return apiadmin.get(`/users/${id}`);
  } catch (error) {
    console.log(error);
  }
};

// Update User Info By Id
export const updateUserInfo = (id, data) => {
  try {
    return apiadmin.put(`/users/${id}`, data);
  } catch (error) {
    console.log(error);
  }
};

// Update User Profile Avatar By Id
export const updateUserAvatar = (id, data) => {
  try {
    return apiadmin.post(`/users/${id}/image`, data);
  } catch (error) {
    console.log(error);
  }
};

