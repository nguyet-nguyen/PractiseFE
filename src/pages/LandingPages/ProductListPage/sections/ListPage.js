import {getAllCategory, getAllProducts, getAllUsers} from "features/Api";
import React, {useState, useEffect, useRef} from 'react';
import CardProduct from "pages/LandingPages/Home/function/CardProduct";
import {requestFilterCategory} from "features/Api";
import Loading from "../../../../Loading";
import {useLocation} from "react-router-dom";
import { numberFormat } from "pages/LandingPages/Home/function/FormatMoney";

const ListPage = ({sidebarOpen, setSidebarOpen, categoryList}) => {
    const location = useLocation();
    const {pathname} = location;

    const trigger = useRef(null);
    const sidebar = useRef(null);

    const storedSidebarExpanded = localStorage.getItem('sidebar-expanded');
    const [sidebarExpanded, setSidebarExpanded] = useState(storedSidebarExpanded === null ? false : storedSidebarExpanded === 'true');

    // close on click outside
    useEffect(() => {
        const clickHandler = ({target}) => {
            if (!sidebar.current || !trigger.current) return;
            if (!sidebarOpen || sidebar.current.contains(target) || trigger.current.contains(target)) return;
            setSidebarOpen(false);
        };
        document.addEventListener('click', clickHandler);
        return () => document.removeEventListener('click', clickHandler);
    });

    // close if the esc key is pressed
    useEffect(() => {
        const keyHandler = ({keyCode}) => {
            if (!sidebarOpen || keyCode !== 27) return;
            setSidebarOpen(false);
        };
        document.addEventListener('keydown', keyHandler);
        return () => document.removeEventListener('keydown', keyHandler);
    });

    useEffect(() => {
        localStorage.setItem('sidebar-expanded', sidebarExpanded);
        if (sidebarExpanded) {
            document.querySelector('body').classList.add('sidebar-expanded');
        } else {
            document.querySelector('body').classList.remove('sidebar-expanded');
        }
    }, [sidebarExpanded]);
    // ----------------------------------
    const [productListFilter, setProductListFilter] = useState([]);
    const [loading, setLoading] = useState(false);
    const [page, setPage] = useState(1);
    const [pageItem, setPageItem] = useState(1);
    // useEffect(() => {
    //     setLoading(true);
    //     requestFilterCategory(data, pageItem, 6)
    //         .then((response) => {
    //             setLoading(false);
    //             setProductListFilter(response.data.data);
    //             if (response.data.total % 6 == 0) {
    //                 setPage(response.data.total / 6);
    //             } else {
    //                 setPage(Math.floor((response.data.total / 6)) + 1);
    //             }
    //         })
    //         .catch((err) => {
    //             setLoading(false)
    //             console.warn(err);
    //         });
    // }, []);
    // -------------pagination--------------
    console.log(page)
    const changePage = (item) => {
        setPageItem(item)
        requestFilterCategory(data ,item, 6)
            .then((response) => {
                setProductListFilter(response.data.data);
            })
            .catch((err) => {
                console.warn(err);
            });
        console.log(item);
    }
    const changePagePrevious = (item) => {
        if (item >= 1) {
            setPageItem(item)
            requestFilterCategory(data ,item, 6)
                .then((response) => {
                    setProductListFilter(response.data.data);
                })
                .catch((err) => {
                    console.warn(err);
                });
        }
        console.log(item);
    }
    const changePageNext = (item) => {
        if (item <= page) {
            setPageItem(item)
            requestFilterCategory(data ,item, 6)
                .then((response) => {
                    setProductListFilter(response.data.data);
                })
                .catch((err) => {
                    console.warn(err);
                });
        }
        console.log(item)
    }

    // ----------------filter-----------------
    const [sort, setSort] = useState("");
    const [cate, setCate] = useState(1);
    const [priceState, setPriceState] = useState(1);
    const [minPrice, setMinPrice] = useState(0);
    const [maxPrice, setMaxPrice] = useState("");
    const [search, setSearch] = useState("");
    const data = {
        sort: sort,
        category: cate,
        minPrice: minPrice,
        maxPrice: maxPrice,
        keyword: search,
    }
    console.log(data);

    const getcategoryId = (e) => {
        setCate(e.target.value);
    }
    const getPriceLevel = (e) => {
        setPriceState(e.target.value);
        if (e.target.value == 1) {
            setMinPrice("");
            setMaxPrice("");
        } else if (e.target.value == 2) {
            setMinPrice(0);
            setMaxPrice(500);
        } else if (e.target.value == 3) {
            setMinPrice(500);
            setMaxPrice(1000);
        } else if (e.target.value == 4) {
            setMinPrice(1000);
            setMaxPrice(3000);
        } else if (e.target.value == 5) {
            setMinPrice(3000);
            setMaxPrice(5000);
        } else if (e.target.value == 6) {
            setMinPrice(5000);
            setMaxPrice("");
        }
    }
    const getSortLevel = (e) => {
        setSort(e.target.value)
    }
    const priceLevel = [
        {
            id: 1
        },
        {
            id: 2,
            minPrice: 0,
            maxPrice: 500,
        },
        {
            id: 3,
            minPrice: 500,
            maxPrice: 1000,
        },
        {
            id: 4,
            minPrice: 1000,
            maxPrice: 3000,
        },
        {
            id: 5,
            minPrice: 3000,
            maxPrice: 5000,
        },
        {
            id: 6,
            minPrice: 5000,
        },
    ];
    const sortLevel = [
        {
            id: 1,
            name: "price-DESC",
            description: "High - Low Price"
        },
        {
            id: 2,
            name: "price-ASC",
            description: "Low - High Price",

        },
        {
            id: 3,
            name: "createdAt-DESC",
            description: "Latest items",
        },
        {
            id: 4,
            name: "createdAt-ASC",
            description: "Oldest items",
        }
    ]
    const searchProduct = (e) => {
        console.log(e.target.value);
        setSearch(e.target.value);

    }
    // setLoading(true);
    useEffect(() => {
        // setLoading(true);
        requestFilterCategory(data, pageItem, 6)
            .then(response => {
            setLoading(false);
            setProductListFilter(response.data.data);
            if (response.data.total % 6 == 0) {
                setPage(response.data.total / 6);
            } else {
                setPage(Math.floor((response.data.total / 6)) + 1);
            }
        }).catch(err => {
            console.log(err);
        })
    })
    return (
        <div id="listPage-productList">
            <div className="grid md:grid-cols-4 grid-cols-1 gap-4">
                <div className="md:mt-14 mt-0">
                    <div className='-left-4'>
                        <div
                            className={`md:fixed inset-0 bg-white bg-opacity-30 z-40 lg:hidden lg:z-auto transition-opacity duration-200 ${sidebarOpen ? 'opacity-100' : 'opacity-0 pointer-events-none'}`}
                            aria-hidden="true"></div>
                        <div
                            id="sidebar"
                            ref={sidebar}
                            className={`flex flex-col items-center justify-center absolute bg-white z-40 left-0 top-0 lg:static lg:left-auto lg:top-auto
                 lg:translate-x-0 transform h-auto overflow-y-scroll lg:overflow-y-auto no-scrollbar w-72 
                 lg:w-20 lg:sidebar-expanded:!w-72 2xl:!w-72 shrink-0 transition-all 
                 duration-200 ease-in-out  ${sidebarOpen ? 'translate-x-0' : '-translate-x-64'}`}>
                            {/* Sidebar header */}
                            <div className="flex justify-between items-center pr-3 sm:px-2 hidden">
                                {/* Close button */}
                                <button
                                    ref={trigger}
                                    className="lg:hidden text-slate-500 hover:text-slate-400"
                                    onClick={() => setSidebarOpen(!sidebarOpen)}
                                    aria-controls="sidebar"
                                    aria-expanded={sidebarOpen}
                                >
                                    <span className="sr-only">Close sidebar</span>
                                    <svg className="w-6 h-6 fill-current" viewBox="0 0 24 24"
                                         xmlns="http://www.w3.org/2000/svg">
                                        <path d="M10.7 18.7l1.4-1.4L7.8 13H20v-2H7.8l4.3-4.3-1.4-1.4L4 12z"/>
                                    </svg>
                                </button>
                            </div>

                            {/* Links */}
                            <div className="border-2 rounded-md border-solid border-zinc-300 p-5">
                                {/* Pages group */}

                                <div>
                                    <h3 className="uppercase text-slate-500 font-bold flex">
                                        <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6 text-amber-700"
                                             fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth={2}>
                                            <path strokeLinecap="round" strokeLinejoin="round"
                                                  d="M5 12h14M5 12a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v4a2 2 0 01-2 2M5 12a2 2 0 00-2 2v4a2 2 0 002 2h14a2 2 0 002-2v-4a2 2 0 00-2-2m-2-4h.01M17 16h.01"/>
                                        </svg>
                                        <span
                                            className="text-amber-700 ml-3 text-lg lg:hidden lg:sidebar-expanded:block 2xl:block uppercase">All Categories</span>
                                    </h3>
                                    <ul className="mt-3">
                                        {categoryList && categoryList.map((category) =>
                                            <li className={`px-3 py-2 rounded-sm mb-0.5 last:mb-0 ${pathname === '/dashboard' && 'bg-slate-900'}`}>
                                                <div className="form-check">
                                                    <input
                                                        checked={cate == category.id}
                                                        onClick={getcategoryId}
                                                        className="form-check-input appearance-none rounded-full h-4 w-4
                                                border border-gray-300 bg-white checked:bg-amber-600
                                                checked:border-amber-600 focus:outline-none
                                                transition duration-200 mt-1
                                                align-top bg-no-repeat bg-center bg-contain float-left mr-2 cursor-pointer"
                                                        type="radio"
                                                        name="category"
                                                        value={category.id}
                                                        id={category.id}
                                                    />
                                                    <label
                                                        className="form-check-label inline-block text-gray-800 capitalize"
                                                        htmlFor="flexCheckDefault"
                                                    >
                                                        {category.name}
                                                    </label>
                                                </div>
                                            </li>
                                        )}
                                    </ul>
                                </div>

                                <div>
                                    <h3 className="uppercase text-slate-500 font-bold flex mt-5">
                                        <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6 text-amber-700"
                                             fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth={2}>
                                            <path strokeLinecap="round" strokeLinejoin="round"
                                                  d="M5 12h14M5 12a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v4a2 2 0 01-2 2M5 12a2 2 0 00-2 2v4a2 2 0 002 2h14a2 2 0 002-2v-4a2 2 0 00-2-2m-2-4h.01M17 16h.01"/>
                                        </svg>
                                        <span
                                            className="text-amber-700 ml-3 text-lg lg:hidden lg:sidebar-expanded:block 2xl:block uppercase">price level</span>
                                    </h3>
                                    <ul className="mt-3">
                                        {priceLevel.map((price, index) =>
                                            <li className={`px-3 py-2 rounded-sm mb-0.5 last:mb-0 ${pathname === '/dashboard' && 'bg-slate-900'}`}>
                                                {index == 0
                                                    ?
                                                    <div className="form-check">
                                                        <input
                                                            checked={priceState == price.id}
                                                            onClick={getPriceLevel}
                                                            className="form-check-input appearance-none rounded-full h-4 w-4
                                                border border-gray-300 bg-white checked:bg-amber-600
                                                checked:border-amber-600 focus:outline-none
                                                transition duration-200 mt-1
                                                align-top bg-no-repeat bg-center bg-contain float-left mr-2 cursor-pointer"
                                                            type="radio"
                                                            name="price"
                                                            value={price.id}
                                                            id={price.id}
                                                        />
                                                        <label
                                                            className="form-check-label inline-block text-gray-800 capitalize"
                                                            htmlFor="flexCheckDefault">
                                                            At all
                                                        </label>
                                                    </div>
                                                    :
                                                    (index == 1 || index == 5) ?
                                                        <div className="form-check">
                                                            <input
                                                                checked={priceState == price.id}
                                                                onClick={getPriceLevel}
                                                                className="form-check-input appearance-none rounded-full h-4 w-4
                                                border border-gray-300 bg-white checked:bg-amber-600
                                                checked:border-amber-600 focus:outline-none
                                                transition duration-200 mt-1
                                                align-top bg-no-repeat bg-center bg-contain float-left mr-2 cursor-pointer"
                                                                type="radio"
                                                                name="price"
                                                                value={price.id}
                                                                id={price.id}
                                                            />
                                                            {index == 1 ?
                                                                <label
                                                                    className="form-check-label inline-block text-gray-800 capitalize"
                                                                    htmlFor="flexCheckDefault">
                                                                    Under {numberFormat(price.maxPrice)}
                                                                </label>
                                                                :
                                                                <label
                                                                    className="form-check-label inline-block text-gray-800 capitalize"
                                                                    htmlFor="flexCheckDefault">
                                                                    Over {numberFormat(price.minPrice)}
                                                                </label>
                                                            }

                                                        </div>
                                                        :
                                                        <div className="form-check">
                                                            <input
                                                                onClick={getPriceLevel}
                                                                className="form-check-input appearance-none rounded-full h-4 w-4
                                                border border-gray-300 bg-white checked:bg-amber-600
                                                checked:border-amber-600 focus:outline-none
                                                transition duration-200 mt-1
                                                align-top bg-no-repeat bg-center bg-contain float-left mr-2 cursor-pointer"
                                                                type="radio"
                                                                name="price"
                                                                value={price.id}
                                                                id={price.id}
                                                            />
                                                            <label
                                                                className="form-check-label inline-block text-gray-800 capitalize"
                                                                htmlFor="flexCheckDefault">
                                                                {numberFormat(price.minPrice)} - {numberFormat(price.maxPrice)}
                                                            </label>
                                                        </div>


                                                }


                                            </li>
                                        )}

                                    </ul>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="col-span-3">
                    <div>
                        {loading ? <Loading/> : ""}
                        <button
                            className="lg:hidden buttonSideNav bg-amber-600 p-2 rounded-r-md
                text-white"
                            aria-controls="sidebar"
                            aria-expanded={sidebarOpen}
                            onClick={() => setSidebarOpen(!sidebarOpen)}
                        >
                            <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24"
                                 stroke="currentColor" strokeWidth={2}>
                                <path strokeLinecap="round" strokeLinejoin="round" d="M4 6h16M4 12h16M4 18h16"/>
                            </svg>
                        </button>
                        <div className="flex justify-between items-center">
                            <h2 className="font-bold text-color leading-tight text-3xl text-center uppercase">All
                                Items
                            </h2>
                            <form className="w-full max-w-sm bg-white">
                                <div className="flex items-center rounded-full border-2 border-amber-600 px-5 py-2">
                                    <input
                                        onBlur={searchProduct}
                                        className="appearance-none bg-transparent border-none w-full
                                    text-gray-700 mr-3 py-1 px-2 leading-tight focus:outline-none
                                    placeholder:text-center"
                                        type="text"
                                        placeholder="Search"
                                        aria-label="Full name"
                                    />
                                </div>
                            </form>

                            <div className="w-52 w-full">
                                <select onChange={getSortLevel}
                                        className="form-select appearance-none block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding bg-no-repeat border border-solid border-gray-300 rounded transition ease-in-out m-0
                        focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none capitalize"
                                        aria-label="Default select example">
                                    {sortLevel.map((sort) =>
                                        <option id={sort.name} value={sort.name}
                                                className="p-2 m-2 capitalize">{sort.description}</option>
                                    )}
                                </select>
                            </div>
                        </div>
                        <div className="grid md:grid-cols-3 grid-cols-2 gap-6 md:mt-6 mt-3">
                            <CardProduct proList={productListFilter}/>
                        </div>
                    </div>
                </div>
            </div>
            <div className="flex justify-end mt-5 mr-4">
                <div>
                    <nav
                        className="relative z-0 inline-flex rounded-md shadow-sm -space-x-px"
                        aria-label="Pagination">
                        <button
                            onClick={() => changePagePrevious(pageItem - 1)}
                            className="relative inline-flex items-center px-2 py-2 rounded-l-md border border-amber-600
                                bg-white text-sm font-medium text-amber-600 hover:text-white hover:bg-amber-600">
                            <span className="sr-only">Previous</span>
                            <i className="fa fa-angle-double-left" aria-hidden="true"></i>
                        </button>
                        {Array.from(Array(page), (e, item) => {
                            return <button
                                onClick={() => changePage(item + 1)}
                                className={`z-10 border-amber-600 relative
                                 inline-flex items-center px-4 py-2 border text-sm font-medium hover:text-white 
                                 hover:bg-amber-600 ${(pageItem == (item + 1)) ? "text-white bg-amber-600" : "text-amber-600 bg-white"}`}>{item + 1}
                            </button>
                        })}
                        <button onClick={() => changePageNext(pageItem + 1)}
                                className="relative inline-flex items-center px-2 py-2 rounded-r-md border border-amber-600
                                bg-white text-sm font-medium text-amber-600 hover:text-white hover:bg-amber-600">
                            <i className="fa fa-angle-double-right" aria-hidden="true"></i>
                        </button>
                    </nav>
                </div>
            </div>
        </div>


    );
}

export default ListPage;