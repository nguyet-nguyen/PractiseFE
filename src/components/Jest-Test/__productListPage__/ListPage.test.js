import { getAllProducts } from "features/Api";
import { faker } from "@faker-js/faker";
import ListPage from "pages/LandingPages/ProductListPage/sections/ListPage";
jest.mock("pages/LandingPages/ProductListPage/sections/ListPage")
describe("Jest Unit Test ListPage", () => {

    it("Test Asynchonous APIs",

        async () => {
            let id = faker.datatype.number();
            let name = faker.name;
            let image = faker.image;
            let category = faker.category
            let price = faker.price; 
            let spyOn = jest.spyOn(ListPage,'getcategoryId').mockResolvedValueOnce(
                [
                    {
                        "id": id,
                        "name": name,
                        "image": image,
                        "category": category,
                        "price": price,
                    }
                ]
            );

            // eslint-disable-next-line jest/valid-expect-in-promise
            await getAllProducts()
                .then(data => {
                    expect(data[0].id).toEqual(id);
                    expect(data[0].name).toEqual(name);
                    expect(data[0].image).toEqual(image);
                    expect(data[0].category).toEqual(category);
                    expect(data[0].price).toEqual(price);
                })

            spyOn.mockRestore();
        })
})