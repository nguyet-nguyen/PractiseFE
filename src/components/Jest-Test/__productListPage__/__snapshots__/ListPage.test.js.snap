// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Unit Test ListPage productListPage SnapShot case 1 1`] = `
<section
  class="container h-auto mx-auto md:px-5 px-0 py-2 mx-auto md:py-12 md:px-32"
  id="listPage-productList"
>
  <div
    class="md:w-12/12 w-full h-auto transition-all rounded-lg mx-auto"
  >
    <h2
      class="font-bold text-color leading-tight text-3xl mb-2 text-center uppercase"
    >
      All Items
    </h2>
    <nav
      class="navbar navbar-expand-lg shadow-lg py-2 bg-gray-50 relative md:flex md:items-center w-full md:justify-between block px-2"
    >
      <div
        class="px-6"
      >
        <button
          aria-controls="navbarSupportedContentX"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler border-0 py-3 lg:hidden leading-none text-xl bg-transparent text-gray-600 hover:text-gray-700 focus:text-gray-700 transition-shadow duration-150 ease-in-out"
          data-bs-target="#navbarSupportedContentX"
          data-bs-toggle="collapse"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="w-5"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="navbar-collapse collapse grow items-center"
          id="navbarSupportedContentX"
        >
          <ul
            class="navbar-nav mr-auto flex flex-row"
          >
            <li
              class="nav-item dropdown static"
            >
              <a
                aria-expanded="false"
                class="nav-link block pr-2 lg:px-2 py-2 text-gray-600 hover:text-gray-700 focus:text-gray-700 transition duration-150 ease-in-out dropdown-toggle flex items-center whitespace-nowrap"
                data-bs-toggle="dropdown"
                data-mdb-ripple="true"
                data-mdb-ripple-color="light"
                href="#"
                id="dropdownMenuButtonX"
                type="button"
              >
                Mega menu
                <svg
                  aria-hidden="true"
                  class="w-2 ml-2"
                  data-icon="caret-down"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    fill="currentColor"
                  />
                </svg>
              </a>
              <div
                aria-labelledby="dropdownMenuButtonX"
                class="dropdown-menu w-full mt-0 hidden shadow-lg bg-white absolute left-0 top-full"
              >
                <div
                  class="px-6 lg:px-8 py-5"
                >
                  <div
                    class="grid md:grid-cols-2 lg:grid-cols-4 gap-6"
                  >
                    <div
                      class="bg-white text-gray-600"
                    >
                      <a
                        aria-current="true"
                        class="block px-6 py-2 border-b border-gray-200 w-full hover:bg-gray-50 hover:text-gray-700 transition duration-150 ease-in-out"
                        href="#!"
                      >
                        Lorem ipsum
                      </a>
                      <a
                        aria-current="true"
                        class="block px-6 py-2 border-b border-gray-200 w-full hover:bg-gray-50 hover:text-gray-700 transition duration-150 ease-in-out"
                        href="#!"
                      >
                        Dolor sit
                      </a>
                      <a
                        aria-current="true"
                        class="block px-6 py-2 border-b border-gray-200 w-full hover:bg-gray-50 hover:text-gray-700 transition duration-150 ease-in-out"
                        href="#!"
                      >
                        Amet consectetur
                      </a>
                      <a
                        aria-current="true"
                        class="block px-6 py-2 border-b border-gray-200 w-full hover:bg-gray-50 hover:text-gray-700 transition duration-150 ease-in-out"
                        href="#!"
                      >
                        Cras justo odio
                      </a>
                      <a
                        aria-current="true"
                        class="block px-6 py-2 w-full hover:bg-gray-50 hover:text-gray-700 transition duration-150 ease-in-out"
                        href="#!"
                      >
                        Adipisicing elit
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="w-52 w-full"
      >
        <select
          aria-label="Default select example"
          class="form-select appearance-none block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding bg-no-repeat border border-solid border-gray-300 rounded transition ease-in-out m-0
 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none capitalize"
        />
      </div>
    </nav>
    <div
      class="grid grid-cols-4 gap-6 md:mt-6 mt-3"
    />
  </div>
</section>
`;
